File: 0350-intersection-of-two-arrays-ii/0350-intersection-of-two-arrays-ii.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        unordered_map<int,int> mpp;
        vector<int> l;

        for(int i=0; i<nums1.size(); i++){
            mpp[nums1[i]]++;
        }
        for(int j=0; j<nums2.size(); j++){
            if (mpp.find(nums2[j]) != mpp.end() && mpp[nums2[j]] != 0){
                l.push_back(nums2[j]);
                mpp[nums2[j]]--;
            }
        }
        return l;
        
    }
};

File: 0350-intersection-of-two-arrays-ii/0350-intersection-of-two-arrays-ii.py
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        d = {}
        l = []
        for i in range(len(nums1)):
            if nums1[i] in d:
                d[nums1[i]] += 1
            else:
                d[nums1[i]] = 1

        for j in range(len(nums2)):
            if nums2[j] in d and d[nums2[j]] != 0:
                l.append(nums2[j])
                d[nums2[j]] -= 1

        return l
        